src/adapter/src/coord/appends.rs

```diff
@@ -352,47 +350,16 @@ impl<S: Append + 'static> Coordinator<S> {
         &mut self,
         timestamp: Timestamp,
         responses: Vec<CompletedClientTransmitter<ExecuteResponse>>,
-        in_ddl: bool,
         _write_lock_guard: Option<OwnedMutexGuard<()>>,
     ) {
         self.apply_local_write(timestamp).await;
-
-        // If we're in the middle of DDL then the catalog and COMPUTE/STORAGE may be out of sync,
-        // so we don't advance other timelines and we don't update read holds.
-        if !in_ddl {
-            let global_timelines = std::mem::take(&mut self.global_timelines);
-            for (
-                timeline,
-                TimelineState {
-                    mut oracle,
-                    mut read_holds,
-                },
-            ) in global_timelines
-            {
-                let now = if timeline == Timeline::EpochMilliseconds {
-                    timestamp
-                } else {
-                    // For non realtime sources, we define now as the largest timestamp, not in
-                    // advance of any object's upper. This is the largest timestamp that is closed
-                    // to writes.
-                    let id_bundle = self.ids_in_timeline(&timeline);
-                    self.largest_not_in_advance_of_upper(&id_bundle)
-                };
-                oracle
-                    .apply_write(now, |ts| self.catalog.persist_timestamp(&timeline, ts))
-                    .await;
-                let read_ts = oracle.read_ts();
-                if read_holds.time.less_than(&read_ts) {
-                    read_holds = self.update_read_hold(read_holds, read_ts).await;
-                }
-                self.global_timelines
-                    .insert(timeline, TimelineState { oracle, read_holds });
-            }
-        }
-
         for response in responses {
             response.send();
         }
+
+        self.internal_cmd_tx
```

Comment:
Some of my comment below is applicable to this as well. We were mainly advancing timelines here out of convenience and not because of any guarantees. 

We explicitly didn't advance timelines during DDL, because group commit happened in the middle of a DDL command so we were in a partial state (where some of the Coord state reflected the DDL and some of the Coord state didn't). Now that the timeline advancement happens after the DDL, instead of the middle, it's fine to run.

I will try and add some of this in a comment.